src/main.rs

```diff
@@ -92,29 +102,62 @@ impl HanoiError {
 ///    * `Action`: if the input was well formed
 ///    * `Hanoi::UnknownCommand`: otherwise
 fn parse_action(input: &str) -> Result<Action,HanoiError> {
-    unimplemented!()
+    if input.len() == 1 {
+        return match input.chars().nth(0).unwrap() {
+            'q' => Ok(Action::Quit),
+            _ => Err(HanoiError::UnknownCommand)
+        };
+    }
+
+    if input.len() != 2 {
+        return Err(HanoiError::UnknownCommand);
+    }
+
+    let from = match parse_peg(input.chars().nth(0).unwrap()) {
+        Result::Ok(value) => value,
+        Result::Err(err) => return Err(err),
+    };
+    let to = match parse_peg(input.chars().nth(1).unwrap()) {
+        Result::Ok(value) => value,
+        Result::Err(err) => return Err(err),
+    };
+
+    match from == to {
+        true => Err(HanoiError::UnknownCommand),
+        false => Ok(Action::Move(Move::new(from, to)))
+    }
 }
 
 impl State {
 
     /// Creates a Towers of Hanoi game with `disks` disks in a single tower
     fn new(disks: u8) -> State {
-        unimplemented!()
+        State{left: (1..disks+1).rev().map(|n| Disk(n)).collect(),
```

Comment:
`.map(Disk)` also works, which is pretty neat :smile:!

Nice job though -- this is really slick!